neuraxle/steps/sklearn.py

```diff
@@ -78,8 +78,15 @@ def transform(self, data_inputs):
             return self.wrapped_sklearn_predictor.predict(data_inputs)
         return self.wrapped_sklearn_predictor.transform(data_inputs)
 
-    def set_hyperparams(self, flat_hyperparams: HyperparameterSamples) -> BaseStep:
-        BaseStep.set_hyperparams(self, flat_hyperparams)
+    def set_hyperparams(self, hyperparams: HyperparameterSamples) -> BaseStep:
+        BaseStep.set_hyperparams(self, hyperparams)
+
+        # this should work
+        flat_hyperparams = {
+            '__'.join(key.split('__')[1:]): value for key, value in HyperparameterSamples(hyperparams).to_flat().items()
+        }
```

Comment:
@guillaume-chevalier I am stuck here. I don't know what to do. The set_params is strange.